src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -106,7 +106,7 @@ private BoundExpression CompareWithConstant(BoundExpression input, BoundExpressi
                 );
         }
 
-        BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input, LocalSymbol target, bool requiresNullTest)
+        BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input, BoundExpression target, bool requiresNullTest)
```

Comment:
This code is not assuming that the target is a local or pseudo-local. In fact, this code has no knowledge or understanding of pseudo-locals.

---
In reply to: [77861412](https://github.com/dotnet/roslyn/pull/13535#discussion_r77861412) [](ancestors = 77861412,77746892,77736440,77735261,77712197)